codec-http/src/main/java/io/netty/handler/codec/http/multipart/HttpPostMultipartRequestDecoder.java

```diff
@@ -805,6 +796,29 @@ private InterfaceHttpData findMultipartDisposition() {
         }
     }
 
+    private static final Pattern DOUBLE_SINGLE_QUOTE = Pattern.compile("''");
+    private static final String FILENAME_ENCODED = HttpHeaderValues.FILENAME.toString() + '*';
+
+    private Attribute getContentDispositionAttribute(String... values) {
+        String name = cleanString(values[0]);
+        String value = values[1];
+
+        // See http://www.w3.org/Protocols/rfc2616/rfc2616-sec19.html
+        if (HttpHeaderValues.FILENAME.contentEquals(name)) {
+            // filename value is quoted string so strip them
+            value = value.substring(1, value.length() - 1);
+        } else if (FILENAME_ENCODED.equals(name)) {
+            // filename value is encoded. See https://tools.ietf.org/html/rfc5987
+            name = name.substring(0, name.length() - 1);
+            String[] split = DOUBLE_SINGLE_QUOTE.split(value, 2);
+            value = QueryStringDecoder.decodeComponent(split[1], Charset.forName(split[0]));
```

Comment:
Here too, you might consider check that the array size is 2 so that you don't throw an NPE.

Reply:
Considered this, but if not an NPE, then what? There are other spots in the code around this PR that split and don't check.